vesmod/EdgeMod/spectrum_utils.py

```
@@ -97,35 +98,67 @@ def fit_spectrum_to_theory_lmfit(fitting_group, lmax, free_sigma=False, weighted
+    ``lmax`` is an exclusive upper summation bound. Public calls require
+    physical Fourier modes q >= 2 and q < lmax.
     """
+    modes = _validate_hss97_inputs(q, kC, sigma, lmax)
     function = []
     lmax = int(lmax)
-    for wavenum in q:
-        wavenum = int(wavenum)
-        summ = 0
+    for wavenum in modes:
+        summ = 0.0
         for l in range(wavenum, lmax):
```

Ruff (0.16.3)

[error] Ambiguous variable name: `l`

(E741)